maths/extended_euclidean_algorithm.py

```diff
@@ -3,69 +3,97 @@
 
 Finds 2 numbers a and b such that it satisfies
 the equation am + bn = gcd(m, n) (a.k.a Bezout's Identity)
+
+https://en.wikipedia.org/wiki/Extended_Euclidean_algorithm
 """
 
 # @Author: S. Sharma <silentcat>
 # @Date:   2019-02-25T12:08:53-06:00
 # @Email:  silentcat@protonmail.com
-# @Last modified by:   PatOnTheBack
-# @Last modified time: 2019-07-05
+# @Last modified by:   pikulet
+# @Last modified time: 2020-10-02
 
 import sys
 
 
-def extended_euclidean_algorithm(m, n):
+def sign(n: int) -> int:
+    """
+    Returns sign of number for correction of negative numbers in algorithm
+
+    >>> sign(4)
+    1
+
+    >>> sign(0)
+    0
+
+    >>> sign(-8)
+    -1
+
+    """
+    if n > 0:
+        return 1
+    elif n < 0:
+        return -1
+    else:
+        return 0
+
+
+def extended_euclidean_algorithm(a: int, b: int) -> (int, int):
     """
     Extended Euclidean Algorithm.
 
     Finds 2 numbers a and b such that it satisfies
     the equation am + bn = gcd(m, n) (a.k.a Bezout's Identity)
+
+    >>> extended_euclidean_algorithm(1, 24)
+    (1, 0)
+
+    >>> extended_euclidean_algorithm(8, 14)
+    (2, -1)
+
+    >>> extended_euclidean_algorithm(240, 46)
+    (-9, 47)
+
+    >>> extended_euclidean_algorithm(1, -4)
+    (1, 0)
+
+    >>> extended_euclidean_algorithm(-2, -4)
+    (-1, 0)
+
+    >>> extended_euclidean_algorithm(0, -4)
+    (0, -1)
+
+    >>> extended_euclidean_algorithm(2, 0)
+    (1, 0)
+
     """
-    a = 0
-    a_prime = 1
-    b = 1
-    b_prime = 0
-    q = 0
-    r = 0
-    if m > n:
-        c = m
-        d = n
-    else:
-        c = n
-        d = m
-
-    while True:
-        q = int(c / d)
-        r = c % d
-        if r == 0:
-            break
-        c = d
-        d = r
-
-        t = a_prime
-        a_prime = a
-        a = t - q * a
-
-        t = b_prime
-        b_prime = b
-        b = t - q * b
-
-    pair = None
-    if m > n:
-        pair = (a, b)
+    old_r, r = a, b
+    old_s, s = 1, 0
+    old_t, t = 0, 1
```

Comment:
Please use descriptive variable names everywhere.

Reply:
I have changed to 

r --> remainder
s--> coeff_a
t --> coeff_b